website_docs/getting-started.md

```diff
@@ -3,552 +3,897 @@ title: "Getting Started"
+                        }
+                }
+        ],
+        "InstrumentationLibrary": {
+                "Name": "diceroller-service",
+                "Version": "",
+                "SchemaURL": ""
+        }
 }
 ```
 
-There's a fair amount going on here. First you are creating a console exporter that will export to a file. You are then registering the exporter with a new [`TracerProvider`]. This is done with a [`BatchSpanProcessor`] when it is passed to the [`trace.WithBatcher`] option. Batching data is a good practice and will help not overload systems downstream. Finally, with the [`TracerProvider`] created, you are deferring a function to flush and stop it, and registering it as the global OpenTelemetry [`TracerProvider`].
+</details>
+
+You'll find that the `dice-roller` span has information about a parent span, and
+the ID of that parent span matches the span created by automatic
+instrumentation.
+
+## Send traces to an OpenTelemetry Collector
+
+The [OpenTelemetry Collector](/docs/collector/getting-started/) is a critical
+component of most production deployments. Some examples of when it's beneficial
+to use a collector:
+
+* A single telemetry sink shared by multiple services, to reduce overhead of
+  switching exporters
+* Aggregating traces across multiple services, running on multiple hosts
+* A central place to process traces prior to exporting them to a backend
+
+Unless you have just a single service or are experimenting, you'll want to use a
+collector in production deployments.
+
+### Configure and run a local collector
+
+First, write the following collector configuration code into `/tmp/`:
+
+```yaml
+# /tmp/otel-collector-config.yaml
+receivers:
+    otlp:
+        protocols:
+            grpc:
+            http:
+exporters:
+    logging:
+        loglevel: debug
+processors:
+    batch:
+service:
+    pipelines:
+        traces:
+            receivers: [otlp]
+            exporters: [logging]
+            processors: [batch]
+```
 
-Do you remember in the previous instrumentation section when we used the global [`TracerProvider`] to get a [`Tracer`]? This last step, registering the [`TracerProvider`] globally, is what will connect that instrumentation's [`Tracer`] with this [`TracerProvider`]. This pattern, using a global [`TracerProvider`], is convenient, but not always appropriate. [`TracerProvider`]s can be explicitly passed to instrumentation or inferred from a context that contains a span. For this simple example using a global provider makes sense, but for more complex or distributed codebases these other ways of passing [`TracerProvider`]s may make more sense.
+Then run the docker command to acquire and run the collector based on this
+configuration:
 
-# Putting It All Together
+```
+docker run -p 4318:4318 \
+    -v /tmp/otel-collector-config.yaml:/etc/otel-collector-config.yaml \
+    otel/opentelemetry-collector:latest \
+    --config=/etc/otel-collector-config.yaml
+```
```

Comment:
Which command did you end up using on your local machine? Here is where I admit my shame: I can't actually get that to work in my ubuntu-via-gitpod environment, but writing to `/tmp` works! 😅

Reply:
I think you should confirm the step works.

Reply:
@hickeyma A few things now that I'm back at it:

* For `/tmp` vs `/etc`, I am following the conventions we have in examples already, e.g. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/examples/tracing/docker-compose.yml#L23. I think it's best to follow that convention, or change it everywhere
* macOS sure is a pain. First, you need to specify a platform explicitly, unlike on linux. Then you can't specify a local path and need an absolute path. If you use `$pwd` then docker freaks out about permissions, and so you need to give it adequate file path permissions and restart docker desktop.

Finally, the following command fails because in the docker image itself, the file is somehow read as a directory instead of a file, and so `ioutil.ReadFile` is literally what's failing:

```sh
docker run -p 4318:4318 \
    -v $pwd/otel-collector-config.yml:/etc/otel-collector-config.yml \
    --platform linux/amd64 \
    otel/opentelemetry-collector:latest \
    --config=/etc/otel-collector-config.yml
```

Same deal if I create a minimal `docker-compose.yml` file like so:

```yaml
version: "3"
services:
  otel-collector:
    image: otel/opentelemetry-collector:latest
    command: ["--config=/etc/otel-collector-config.yml"]
    volumes:
      - ./otel-collector-config.yml:/etc/otel-collector-config.yml
    ports:
      - "4317"
```

(there's no need for the `$pwd` with docker-compose, it seems)

I don't know why, but if I instead just the same file but have it sit in my `/tmp` folder, then everything works. Would appreciate some help here if you've got a command that works locally and doesn't involve writing the config file to `/tmp`.

Reply:
Okay, _actually_, with the docker compose bit this was just me mucking it up.

```yaml
version: "3"
services:
  otel-collector:
    image: otel/opentelemetry-collector:latest
    command: ["--config=/etc/otel-collector-config.yaml"]
    volumes:
      - ./otel-collector-config.yaml:/etc/otel-collector-config.yaml
    ports:
      - "4317"
```

This works now with `docker compose up`. Will see if there's a way to get an equivalent docker command? What I'm trying still interprets the config file as a directory, maybe there's something else the compose tool does. Feels a bit unnecessary to require a compose file for something this simple.

Reply:
On Docker Desktop v4.9 (macOS, apple silicon) the command as-written works correctly. Volume mapping /tmp to /etc in the container is pretty conventional for other documented uses of the otel collector, and writing/reading to /tmp shouldn't require escalation (outside of security hardening issues that we can't be aware of -- I have a fairly stock Mac, and the command works OOB).

I think this is fine as-written and shouldn't block the merge.

Reply:
Its working for me now on Ubuntu. Seemed to have issues using Mac as didn't have Docker Desktop. Thanks.